sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/jsonschema/JsonSchemaResource.java

```diff
@@ -0,0 +1,147 @@
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.UriInfo;
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.eclipse.microprofile.openapi.annotations.media.Content;
+import org.eclipse.microprofile.openapi.annotations.media.ExampleObject;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameter;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameters;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponses;
+import org.eclipse.microprofile.openapi.annotations.security.SecurityRequirement;
+import org.eclipse.microprofile.openapi.annotations.tags.Tag;
+import org.jboss.resteasy.reactive.RestResponse;
+
+/** JSON Schema resource. */
+@Path(JsonSchemaResource.BASE_PATH)
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+@SecurityRequirement(name = OpenApiConstants.SecuritySchemes.TOKEN)
+@Tag(ref = OpenApiConstants.Tags.JSON_SCHEMAS)
+public class JsonSchemaResource {
+
+  public static final String BASE_PATH =
+      "/v2/namespaces/{namespace:\\w+}/collections/{collection:\\w+}/json-schema";
+
+  @Inject @Authorized TableManager tableManager;
+
+  @Inject JsonSchemaManager jsonSchemaManager;
+
+  @Operation(
+      description =
+          "Assign a JSON schema to a collection. This will erase any schema that already exists.")
+  @Parameters(
+      value = {
+        @Parameter(
+            name = "namespace",
+            ref = OpenApiConstants.Parameters.NAMESPACE,
+            description = "The namespace of the collection"),
+        @Parameter(
+            name = "collection",
+            ref = OpenApiConstants.Parameters.COLLECTION,
+            description = "The collection to assign a JSON schema to.")
+      })
+  @APIResponses(
+      value = {
+        @APIResponse(responseCode = "200", description = "Call successful."),
+        @APIResponse(
+            responseCode = "400",
+            description = "Bad request.",
+            content =
+                @Content(
+                    examples = {@ExampleObject(ref = "Invalid JSON schema.")},
+                    schema =
+                        @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                            implementation = ApiError.class))),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_400),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_401),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_500),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_503),
+      })
+  @PUT
+  public Uni<RestResponse<Object>> putJsonSchema(
+      @PathParam("namespace") String namespace,
+      @PathParam("collection") String collection,
+      @NotNull(message = "json schema not provided") JsonNode body) {
+    return jsonSchemaManager
+        .attachJsonSchema(namespace, collection, body)
```

Comment:
confused :confused:

* why are we not getting the valid collection table here?
* why is attach json schema not accepting the table?

Reply:
anyway should be handling missing keyspace/table like:

```
tableManager.getValidDocumentTable(..)
    .onItem()
    .transformToUni(ignored -> jsonSchemaManager.attachJsonSchema(namespace, collection, body))
```

Or alternately, we could use `ensureValidDocumentTable` to create table if not existing, but not sure if this is a breaking change then?

Reply:
I think this is the only method that didn't accept a table, from the other PR - is it necessary to grab the table and then issue CQL afterwards? I guess that would help with authz and stuff, huh....I'll change it, since this isn't a high-traffic endpoint anyway